interfaces/Portalicious/src/app/domains/domain-api.service.ts

```diff
@@ -30,55 +30,71 @@ export abstract class DomainApiService {
   >({
     path,
     processResponse,
-    requestOptions = {},
+    params = {},
+    responseAsBlob = false,
     method = 'GET',
     paginateQuery,
     ...opts
   }: {
     path: Parameters<typeof DomainApiService.prototype.pathToQueryKey>[0];
     processResponse?: (data: BackendDataShape) => ProcessedResponseShape;
-    requestOptions?: Omit<
-      Perform121ServiceRequestParams,
-      'endpoint' | 'method'
+    params?: Record<
+      string,
+      | boolean
+      | number
+      | readonly (boolean | number | string)[]
+      | Signal<
+          boolean | number | readonly (boolean | number | string)[] | string
+        >
+      | string
     >;
+    responseAsBlob?: boolean;
     method?: Perform121ServiceRequestParams['method'];
     paginateQuery?: Signal<PaginateQuery>;
   } & Partial<UndefinedInitialDataOptions<ProcessedResponseShape>>) {
     return () => {
       const queryKey = this.pathToQueryKey(path);
       const endpoint = queryKey.join('/');
+      const deSignalizedParams = Object.fromEntries(
```

Comment:
Moving this here is the main gist of the fix. Without "designalising" the parameters _before_ defining the query key, tanstack query does not detect the changes in the query key.